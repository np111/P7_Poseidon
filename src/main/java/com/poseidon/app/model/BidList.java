package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class BidList {
    private Long id;
    private String account;
    private String type;
    private Double bidQuantity;
    private Double askQuantity;
    private Double bid;
    private Double ask;
    private String benchmark;
    private ZonedDateTime bidListDate;
    private String commentary;
    private String security;
    private String status;
    private String trader;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
