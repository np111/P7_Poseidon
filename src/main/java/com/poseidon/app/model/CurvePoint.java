package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class CurvePoint {
    private Long id;
    private Long curveId;
    private ZonedDateTime asOfDate;
    private Double term;
    private Double value;
    private ZonedDateTime creationDate;
}
