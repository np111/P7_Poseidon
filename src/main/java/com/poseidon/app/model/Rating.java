package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Rating {
    private Long id;

    @NotBlank(message = "MoodysRating is mandatory")
    private String moodysRating;

    @NotBlank(message = "SandPRating is mandatory")
    private String sandPRating;

    @NotBlank(message = "FitchRating is mandatory")
    private String fitchRating;

    @NotNull(message = "Order is mandatory")
    private Byte orderNumber;
}
