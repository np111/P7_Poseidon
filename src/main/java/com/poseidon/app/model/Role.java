package com.poseidon.app.model;

public enum Role {
    USER,
    ADMIN,
}
