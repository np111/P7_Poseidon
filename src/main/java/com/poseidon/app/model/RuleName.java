package com.poseidon.app.model;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class RuleName {
    private Long id;
    private String name;
    private String description;
    private String json;
    private String template;
    private String sqlStr;
    private String sqlPart;
}
