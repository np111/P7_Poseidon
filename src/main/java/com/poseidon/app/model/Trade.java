package com.poseidon.app.model;

import java.time.ZonedDateTime;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Trade {
    private Long id;

    @NotBlank(message = "Account is mandatory")
    private String account;

    @NotBlank(message = "Type is mandatory")
    private String type;

    @NotNull(message = "Buy Quantity is mandatory")
    @DecimalMin(value = "0", inclusive = false, message = "Buy Quantity must be positive")
    private Double buyQuantity;

    private Double sellQuantity;
    private Double buyPrice;
    private Double sellPrice;
    private ZonedDateTime tradeDate;
    private String security;
    private String status;
    private String trader;
    private String benchmark;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
