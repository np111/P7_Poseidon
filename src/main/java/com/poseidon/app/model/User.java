package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Pattern;
import javax.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class User {
    private Long id;

    @NotBlank(message = "Username is mandatory")
    private String username;

    @NotBlank(message = "Password is mandatory")
    @Size(min = 8, message = "Password must be at least 8 characters long")
    @Pattern(regexp = ".*\\p{Lu}.*", message = "Password must contain at least one uppercase letter.")
    @Pattern(regexp = ".*[0-9].*", message = "Password must contain at least one number.")
    @Pattern(regexp = ".*[^\\p{L}0-9].*", message = "Password must contain at least one symbol.")
    private String password;

    @NotBlank(message = "Full Name is mandatory")
    private String fullname;

    @NotNull(message = "Role is mandatory")
    private Role role;
}
