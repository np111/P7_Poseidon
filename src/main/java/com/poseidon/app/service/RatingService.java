package com.poseidon.app.service;

import com.poseidon.app.model.Rating;
import com.poseidon.app.persistence.entity.RatingEntity;
import com.poseidon.app.persistence.mapper.RatingMapper;
import com.poseidon.app.persistence.repository.RatingRepository;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor(onConstructor = @__(@Autowired))
@Service
@Scope("singleton")
public class RatingService extends CrudService<Rating, RatingEntity, Long> {
    private final @Getter RatingRepository repository;
    private final @Getter RatingMapper mapper;

    @Override
    protected void setModelId(Rating user, Long id) {
        user.setId(id);
    }
}
