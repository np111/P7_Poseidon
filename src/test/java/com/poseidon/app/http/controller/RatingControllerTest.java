package com.poseidon.app.http.controller;

import com.poseidon.app.model.Rating;
import com.poseidon.app.persistence.entity.RatingEntity;
import com.poseidon.app.service.RatingService;
import lombok.Getter;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@WebMvcTest(RatingController.class)
class RatingControllerTest extends AbstractCrudControllerTest<RatingService, Rating, RatingEntity, Long> {
    @MockBean
    private @Getter RatingService service;

    public RatingControllerTest() {
        super("/rating", Rating.class, Long.class);
    }

    @Override
    protected Rating getModelA() {
        return Rating.builder()
                .id(1L)
                .moodysRating("moodysRating")
                .sandPRating("sandPRating")
                .fitchRating("fitchRating")
                .orderNumber((byte) 2)
                .build();
    }

    @Override
    protected Rating getModelB() {
        return Rating.builder()
                .id(73582L)
                .moodysRating("pvxupSXofJ")
                .sandPRating("62EDwMlxVa")
                .fitchRating("SU2tecC93S")
                .orderNumber((byte) 58)
                .build();
    }
}
