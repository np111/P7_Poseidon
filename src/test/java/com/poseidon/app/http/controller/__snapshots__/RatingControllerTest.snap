com.poseidon.app.http.controller.RatingControllerTest.add=[
-
> GET /rating/add
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Rating</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/rating/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"moodysRating\" class=\"col-sm-2 control-label\">MoodysRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"moodysRating\" placeholder=\"MoodysRating\" class=\"col-4\" name=\"moodysRating\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sandPRating\" class=\"col-sm-2 control-label\">SandPRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sandPRating\" placeholder=\"SandPRating\" class=\"col-4\" name=\"sandPRating\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"fitchRating\" class=\"col-sm-2 control-label\">FitchRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fitchRating\" placeholder=\"FitchRating\" class=\"col-4\" name=\"fitchRating\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"orderNumber\" class=\"col-sm-2 control-label\">Order</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"orderNumber\" placeholder=\"Order\" class=\"col-4\" name=\"orderNumber\" value=\"\">"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/rating/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add rating\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RatingControllerTest.delete=[
-
> GET /rating/delete/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /rating/delete/73582
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /rating/list
""
]


com.poseidon.app.http.controller.RatingControllerTest.list=[
-
> GET /rating/list
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <div class=\"col-6\">"
"      <a href=\"/bidList/list\">Bid List</a>&nbsp;|&nbsp;"
"      <a href=\"/curvePoint/list\">Curve Points</a>&nbsp;|&nbsp;"
"      <a href=\"/rating/list\">Ratings</a>&nbsp;|&nbsp;"
"      <a href=\"/trade/list\">Trade</a>&nbsp;|&nbsp;"
"      <a href=\"/ruleName/list\">Rule</a>"
"    </div>"
"    <div class=\"col-6 text-right\">"
"      Logged in user: <b class=\"user\"> test-username </b>"
"      <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"        <input type=\"submit\" value=\"Logout\"/>"
"      </form>"
"    </div>"
"  </div>"
"  <div class=\"row\"><h2>Rating List</h2></div>"
"  <div class=\"row\">"
"    <a href=\"/rating/add\" class=\"btn btn-primary btn-sm\">Add New</a>"
"    <table class=\"table table-bordered\">"
"      <thead>"
"      <tr>"
"        <th>Id</th>"
"        <th>MoodysRating</th>"
"        <th>SandPRating</th>"
"        <th>FitchRating</th>"
"        <th>Order</th>"
"        <th>Action</th>"
"      </tr>"
"      </thead>"
"      <tbody>"
"      <tr>"
"        <td>1</td>"
"        <td>moodysRating</td>"
"        <td>sandPRating</td>"
"        <td>fitchRating</td>"
"        <td>2</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/rating/update/1\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/rating/delete/1\">Delete</a>"
"        </td>"
"      </tr>"
"      <tr>"
"        <td>73582</td>"
"        <td>pvxupSXofJ</td>"
"        <td>62EDwMlxVa</td>"
"        <td>SU2tecC93S</td>"
"        <td>58</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/rating/update/73582\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/rating/delete/73582\">Delete</a>"
"        </td>"
"      </tr>"
"      </tbody>"
"    </table>"
"  </div>"
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RatingControllerTest.showUpdate=[
-
> GET /rating/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /rating/update/73582
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update Rating</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/rating/update/73582\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"moodysRating\" class=\"col-sm-2 control-label\">MoodysRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"moodysRating\" placeholder=\"MoodysRating\" class=\"col-4\" name=\"moodysRating\" value=\"pvxupSXofJ\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sandPRating\" class=\"col-sm-2 control-label\">SandPRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sandPRating\" placeholder=\"SandPRating\" class=\"col-4\" name=\"sandPRating\" value=\"62EDwMlxVa\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"fitchRating\" class=\"col-sm-2 control-label\">FitchRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fitchRating\" placeholder=\"FitchRating\" class=\"col-4\" name=\"fitchRating\" value=\"SU2tecC93S\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"orderNumber\" class=\"col-sm-2 control-label\">Order</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"orderNumber\" placeholder=\"Order\" class=\"col-4\" name=\"orderNumber\" value=\"58\">"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/rating/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update rating\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.RatingControllerTest.update=[
-
> POST /rating/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Update Rating</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/rating/update/1\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"moodysRating\" class=\"col-sm-2 control-label\">MoodysRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"moodysRating\" placeholder=\"MoodysRating\" class=\"col-4\" name=\"moodysRating\" value=\"\">"
"          <p class=\"text-danger\">MoodysRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sandPRating\" class=\"col-sm-2 control-label\">SandPRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sandPRating\" placeholder=\"SandPRating\" class=\"col-4\" name=\"sandPRating\" value=\"\">"
"          <p class=\"text-danger\">SandPRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"fitchRating\" class=\"col-sm-2 control-label\">FitchRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fitchRating\" placeholder=\"FitchRating\" class=\"col-4\" name=\"fitchRating\" value=\"\">"
"          <p class=\"text-danger\">FitchRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"orderNumber\" class=\"col-sm-2 control-label\">Order</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"orderNumber\" placeholder=\"Order\" class=\"col-4\" name=\"orderNumber\" value=\"\">"
"          <p class=\"text-danger\">Order is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/rating/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update rating\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /rating/update/1
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /rating/list
""
]


com.poseidon.app.http.controller.RatingControllerTest.validate=[
-
> POST /rating/validate
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Rating</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/rating/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"moodysRating\" class=\"col-sm-2 control-label\">MoodysRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"moodysRating\" placeholder=\"MoodysRating\" class=\"col-4\" name=\"moodysRating\" value=\"\">"
"          <p class=\"text-danger\">MoodysRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"sandPRating\" class=\"col-sm-2 control-label\">SandPRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"sandPRating\" placeholder=\"SandPRating\" class=\"col-4\" name=\"sandPRating\" value=\"\">"
"          <p class=\"text-danger\">SandPRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"fitchRating\" class=\"col-sm-2 control-label\">FitchRating</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"fitchRating\" placeholder=\"FitchRating\" class=\"col-4\" name=\"fitchRating\" value=\"\">"
"          <p class=\"text-danger\">FitchRating is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"orderNumber\" class=\"col-sm-2 control-label\">Order</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"orderNumber\" placeholder=\"Order\" class=\"col-4\" name=\"orderNumber\" value=\"\">"
"          <p class=\"text-danger\">Order is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/rating/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add rating\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /rating/validate
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /rating/list
""
]